pkg/lib/cockpit.js

```diff
@@ -2391,29 +2391,28 @@ function factory() {
 
     let the_user = null;
     cockpit.user = function () {
-        const dfd = cockpit.defer();
-        if (!the_user) {
-            const dbus = cockpit.dbus(null, { bus: "internal" });
-            dbus.call("/user", "org.freedesktop.DBus.Properties", "GetAll",
-                      ["cockpit.User"], { type: "s" })
-                .then(([user]) => {
-                    the_user = {
-                        id: user.Id.v,
-                        name: user.Name.v,
-                        full_name: user.Full.v,
-                        groups: user.Groups.v,
-                        home: user.Home.v,
-                        shell: user.Shell.v
-                    };
-                    dfd.resolve(the_user);
-                })
-                .catch(ex => dfd.reject(ex))
-                .finally(() => dbus.close());
-        } else {
-            dfd.resolve(the_user);
-        }
-
-        return dfd.promise;
+        return new Promise((resolve, reject) => {
```

Comment:
We don't need this extra promise wrapper at all -- it could just directly return the `dbus.call()` promise (chained with .finaly for dbus.close()).